backend/src/test/java/com/google/sps/TestEmailNotifier.java

```diff
@@ -0,0 +1,109 @@
+package com.google.sps;
+
+import com.google.sps.notifs.EmailNotifier;
+import org.junit.Assert;
+import org.junit.Test;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import java.io.IOException;
+import java.util.Properties;
+import static org.junit.Assert.*;
+
+public class TestEmailNotifier {
+
+  public MimeMessage createTestMessage() {
+    Properties properties = System.getProperties();
+    Session session = Session.getDefaultInstance(properties);
+    return new MimeMessage(session);
+  }
+
+  private String getTextFromMessage(MimeMessage message) throws MessagingException, IOException {
+    String result = "";
+    result = message.getContent().toString();
+    return result;
+  }
+
+  @Test
+  public void constructorShouldSetNameAndEmailTo() {
+    EmailNotifier emailNotifier = new EmailNotifier("John", "jdoe@gmail.com", null);
+    assertEquals("John", emailNotifier.getRecipientName());
+    assertEquals("jdoe@gmail.com", emailNotifier.getToEmail());
```

Comment:
Continuing from my comment before: 100% agree that we should test that the email addresses specified int he constructor are used. One way we could do that is test that the emails generated from the class have the right to and from email addresses. Then we can take out these two methods that expose internals of the class.